### PR TITLE
Small cleanup work done during my initial visits to the code.

### PR DESCRIPTION
**Pull request by JamesIry**
These are just a series of small cleanups I did while reading
the code base during my first few days. None are actual
functionality bugs and none warrant a full blown bug. Now
that I'm moving on to doing real work small stuff like this will
likely be swept up in other bugs, so this commit just captures
that initial pass of stuff.

### CODE REVIEW

**Comment by adriaanm** [comment]
LGTM

File: src/compiler/scala/tools/nsc/transform/Mixin.scala [hunk 1]
```diff
@@ -381,20 +381,20 @@ abstract class Mixin extends InfoTransform with ast.TreeDSL {
...
           clazz setFlag lateMODULE
           var sourceModule = clazz.owner.info.decls.lookup(sym.name.toTermName)
-          if (sourceModule != NoSymbol) {
-            sourceModule setPos sym.pos
-            if (sourceModule.flags != MODULE) {
-              log("!!! Directly setting sourceModule flags from %s to MODULE".format(sourceModule.flagString))
-              sourceModule.flags = MODULE
-            }
-          }
-          else {
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/transform/Mixin.scala:391]: For future reference, see Symbol#orElse.

File: src/compiler/scala/tools/nsc/transform/Mixin.scala [hunk 2]
```diff
@@ -381,20 +381,20 @@ abstract class Mixin extends InfoTransform with ast.TreeDSL {
...
             sourceModule = (
               clazz.owner.newModuleSymbol(sym.name.toTermName, sym.pos, MODULE)
                 setModuleClass sym.asInstanceOf[ClassSymbol]
             )
             clazz.owner.info.decls enter sourceModule
           }
+          else {
+            sourceModule setPos sym.pos
+            if (sourceModule.flags != MODULE) {
+              log("!!! Directly setting sourceModule flags from %s to MODULE".format(sourceModule.flagString))
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/transform/Mixin.scala:394]: This should be a devWarning, I wonder how I missed it.

**Comment by paulp** [comment]
LGTM.

**Pull request merged**